xskillscore/core/probabilistic.py

```diff
@@ -475,6 +478,81 @@ def threshold_brier_score(
         return res.mean(dim, keep_attrs=keep_attrs)
 
 
+def _check_bin_dim(ds):
+    """Assert that bin_dim is in ds. Try to guess and rename edges dimension."""
+    bin_dim = "category_edge"
+    if bin_dim not in ds.dims:
+        raise ValueError(f"require {bin_dim} dimension, found {ds.dims}")
+    return ds
+
+
+def _check_data_within_edges(forecasts, forecasts_edges):
+    """Check that forecasts_edges has category_edge dimension and forecasts_edges span range of forecasts (per variable if xr.Dataset)."""
+    if "category_edge" not in forecasts_edges.dims:
+        raise ValueError(
+            f"Expect to find category_edge in forecast_edges, found {forecasts_edges.dims}"
+        )
+
+    def _check(forecasts, forecasts_edges):
```

Comment:
Trying to think if there is a better name than `_check`. I think you've given it this name as it's inside a function and want a function that is only relevant in `_check_data_within_edges`. Given it's a helper to raise errors, could call it something like `raise_value_error_edges` and move it outside the function.

Reply:
I dont want to move it outside this function because the outer function just applies this check to different xr data structures.

Reply:
renamed